src/sorted_vec.h

```diff
@@ -0,0 +1,93 @@
+#ifndef SORTED_SET_H
+#define SORTED_SET_H
+
+#include <algorithm>
+#include <cassert>
+#include <vector>
+
+/// Modifying a key in associative containers leads to UB. This is an std::set
+/// reimplementation with worse characteristics (such as re-sorting the
+/// container on every push), but one that allows for elements to be
+/// out-of-order (which can happen after a key was modified)
+template<class Elem, class Compare = std::less<Elem> >
+class SortedVec {
+  vector<Elem> data;
+  const Compare comp = Compare();
+
+public:
+  size_t size() const {
+    return data.size();
+  }
+
+  void insert(const Elem& elem) {
+    data.push_back(elem);
+    sort(data.begin(), data.end(), comp);
```

Comment:
This doesn't check for uniquness.

Reply:
Okay. Initially I did the check *(the wrong comment you found above leaked from that time)*, but then I wondered, what wrong could happen if I have a duplicate… But if you think something can happen, sure, I'll add the check.

Reply:
> [...] but then I wondered, what wrong could happen if I have a duplicate… But if you think something can happen, [...]

I also don't know^^ Since you named it SortedVec, you're right: It shouldn't check for uniqueness.

Reply:
FTR, I tested it: adding a duplicate results in [this assertion](https://github.com/ninja-build/ninja/blob/20b30dac6698d119e7797b34d6ed2c4ed8f48417/src/build.cc#L100). Okay, I'm gonna make SortedVec a SortedSet then.